### PR TITLE
implement nwp model loading

### LINKED ISSUE #10: benchmark NWP based forecasts
Issue for tracking general info related to benchmark forecasts.

Incomplete list of data availability:

Model | Grid | Source | Total Cloud Cover | GHI | DNI/DHI | Period of Record
-- | -- | -- | -- | -- | -- | --
GFS | 0.5 deg | NOMADS TDS | 3/6 hr mixed interval average | 3/6 hr mixed interval average | None | -2 years to present
GFS | 0.5 deg | Unidata TDS | 3/6 hr mixed interval average | 3/6 hr mixed interval average | None | -30 days to present
NAM | 12 km | NOMADS TDS | 1 (3) hr for 36 (84) hr | 1 (3) hr for 36 (84) hr | None | -1 year to present
NAM | 12 km |   | 1 (3) hr for 36 (84) hr | 1 (3) hr for 36 (84) hr | None | -30 days to present
RAP | 13 km | NOMADS TDS | 1 hr | None | None | -1 year to present
RAP | 13 km | Unidata TDS | 1 hr | None | None | -30 days to present
HRRR | 3 km | Unidata TDS | 1 hr | None | None | -4 days to present
HRRR | 3 km | NCEP DODS | 1 hr | 1 hr | 1 hr | -1 days to present
HRRR | 3 km | NCEP g2sub | None | 15 min | 15 min | -1 days to present

List of methods for converting different inputs into different outputs:

Input | Method | Output (Power)
-- | -- | --
15 min GHI, DNI | I -> P | 15 min or hr avg
15 min GHI | I -> DISC -> P | 15 min or hr avg
1 hr GHI | I -> DISC -> P | 1 hr
1 hr GHI | I -> CSI -> sub-hrly CSI -> I -> DISC -> P | 1 hr avg
1 hr CC | CC -> sub-hrly CC -> Larson -> P | 1 hr avg
3/6 GHI | X | X
3/6 CC | 3/6 CC -> sub-hrly CC -> Larson -> P | 1 hr avg
3/6 CC | 3/6 CC -> 3 hr CC -> sub-hrly CC -> Larson -> P | 1 hr avg

### PR DESCRIPTION
**Pull request by wholmgren**
Closes #10 

Partially addresses #93.

To do:

- [x] Implement or xfail GFS unmix intervals
- [x] fix lat/lon issues in HRRR, RAP, NAM
- [x] xfail or remove 3 hr GFS
- [x] Implement, ignore, or create issue to track more robust file loading?
- [x] Implement or create issue to track more robust testing of values for each model run

### CODE REVIEW

File: solarforecastarbiter/io/nwp.py [hunk 1]
```diff
@@ -41,4 +59,93 @@ def load_forecast(latitude, longitude, init_time, start, end, model,
...
+                         method='nearest')
+        except ValueError:
+            iy_min, ix_min = tunnel_fast(
+                ds.latitude.values, ds.longitude.values,
+                latitude, longitude, limit=limit)
+            pnt = ds.sel(y=iy_min, x=ix_min, method='nearest')
+        else:
+            # we already have the point, but use tunnel_fast to determine
+            # if it's close enough
+            tunnel_fast(ds.latitude.values[:, np.newaxis],
```
Comment (alorenzo175) [inline solarforecastarbiter/io/nwp.py]: why newaxis here but not above? Do you mean to select lat/lon at the point already found? If not, why not just use tunnel_fast for both options?

Comment (wholmgren) [reply solarforecastarbiter/io/nwp.py]: the reason was to ensure that the point was within `limit` of the nearest grid box. new code does the same thing, but hopefully more clear.

**Comment by wholmgren** [comment]
tests fail due to dev api issue. ok to ignore?

**Comment by alorenzo175** [comment]
better now

**Pull request merged**